### PR TITLE
RGB_MATRIX implementation for Novelkeys NK65 (rev < 1.4)

### PR DESCRIPTION
**Pull request by jjournet**
## Description

Implementation of RGB_MATRIX for the Novelkeys NK65 (rev < 1.4)

I have a Novelkeys NK65 for almost 2 years, and even if I created one or two layers, I didn't push QMK too far. Last week, I wanted to implement a RGB color change for each layer, and I discovered that the current implementation of the RGB matrix on the NK65 doesn't follow the 'standard' RGB_MATRIX.

I started playing around , and I have a version of QMK firmware for NK65 that implement RGB_MATRIX. 
(Implement keyboard novelkeys/nk65rgb in parallel of novelkeys/nk65)

All effects seem to work correctly, all keys are working as expected. There is one issue: I realized when testing that my LED on key Column 15 Row 3 (the middle key on the right column, on the right of enter) doesn't work (the key work, not the led). So I don't have the correct key number in the key/led mapping in ```g_led_config``` in ```nk65rgb.c```. Anyone willing to try on it's keyboard and share the correct code is welcome. From the original code and what I could check in different PCB info, I think I have the correct LED number.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — CHANGES_REQUESTED** [review]
We wouldn't want this as a separately supported keyboard.

The changes should be integrated to the existing keyboard files. The main issue, deciding what the defaults would be. Generally as this is targeting `master` we would want core RGB Matrix to be opt-in, though this could change if it were to go via `develop`.

**Comment by jjournet** [comment]
ok, it make sense. I contacted @yiancar but he is on vacation. He is the original maintainer (and designer of the board), I would prefer to wait for his feedback.

**Comment by yiancar** [comment]
Hello I will give the necessary info when I am back from holiday @jjournet but I do agree that however you make the chance it should be in a way that is opt-in as all the other PCBs that have been shipped do not use RGB matrix and will not align with their VIA configuration.

**Comment by jjournet** [comment]
@zvecr following Yiancar comment, and I tend to agree, how would it be possible to provide this alternative implementation for an existing keyboard?

**Comment by drashna** [comment]
I agree with zverc here, this shouldn't be a separate board. 

The best solution right now would be to move all of the wt code (eg `SRC += ...`) into the via keymap's rules.mk, enable RGB Matrix at the keyboard level, and disable it in the via keymap. 

This way, the board supports the wt code ONLY in the via keymap, without breaking the existing via config. 


However, long term, the wt code should be removed completely.

**Comment by jjournet** [comment]
I can try and implement it as an alternate layout, I didn't know it was an option actually.
I just would like to avoid loosing this, as I think it's a good thing to have the RGB_MATRIX available one way or another

File: keyboards/novelkeys/nk65rgb/nk65rgb.h
Comment (drashna) [inline keyboards/novelkeys/nk65rgb/nk65rgb.h:21]:
```suggestion
```

File: keyboards/novelkeys/nk65rgb/readme.md
Comment (drashna) [inline keyboards/novelkeys/nk65rgb/readme.md:2]:
```suggestion
# NK65

```

File: keyboards/novelkeys/nk65rgb/rules.mk
Comment (drashna) [inline keyboards/novelkeys/nk65rgb/rules.mk:36]: These shouldn't be needed
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/novelkeys/nk65rgb/nk65rgb.c
Comment (drashna) [inline keyboards/novelkeys/nk65rgb/nk65rgb.c:184]: NTS: https://github.com/qmk/qmk_firmware/pull/18450

File: keyboards/novelkeys/nk65rgb/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/novelkeys/nk65rgb/keymaps/default/keymap.c:23]:
```suggestion
enum custom_layers {
    BASE,
    ALT,
    RGBL,
};
```

File: keyboards/novelkeys/nk65rgb/rules.mk
Comment (drashna) [inline keyboards/novelkeys/nk65rgb/rules.mk:3]: Is this actually using a proton C, or just a STM32F303?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/novelkeys/nk65rgb/nk65rgb.c
Comment (drashna) [inline keyboards/novelkeys/nk65rgb/nk65rgb.c:190]:
```suggestion
bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
        return false;
    }
    if (host_keyboard_led_state().caps_lock) {
        rgb_matrix_set_color(70, 0, 255, 0 );
    } else {
        rgb_matrix_set_color(70,0, 0, 0 );
    }
    return true;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
